math/exp.c

```diff
@@ -22,9 +22,13 @@
 double frexp(double x, int *exp)
 {
 	conv_t *conv = (conv_t *)&x;
-
 	*exp = 0;
 
+	if (isnan(x) != 0) {
+		errno = EDOM;
+		return NAN;
```

Comment:
In case `isnan(x)` or `isinf(x)` hold, `*exp` can be left unmodified (its value is unspecified). Maybe it's worth to move these above `*exp = 0`, if no legacy test suites / applications rely on the value (glibc?).